forem/forem
# Support devcontainers and GitHub codespace

## What type of PR is this? (check all applicable)

## Description
This is a MVP implementation of devcontainers which is a development tool for vs code and also used on Github Code space!

I would like folks to try it out and let me know what the process feels like. 

Thorough code review not required but appreciated.

## Related Tickets & Documents
Resolves https://github.com/forem/forem-internal-eng/issues/637
Resolves https://github.com/forem/forem-internal-eng/issues/638

## QA Instructions, Screenshots, Recordings


You can try this out in two ways.
Try it out by clicking the button of the upper right corner. This may take a minute. Once complete, you see the server running and pop-up provided to open up port 3000.
![image](https://github.com/forem/forem/assets/15793250/2ed1538a-1d4f-4e12-911d-02771e752023)

And try it out locally!

1. You will need [docker](https://www.docker.com/), VS Code, and vscode's [devcontainer extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers). Apple Silicon users please choose the Apple chip variant.
1. Apple Silicon users: please consider [enabling Rosetta ](https://docs.docker.com/desktop/settings/mac/#use-rosetta-for-x86amd64-emulation-on-apple-silicon)for faster process time. Restart may be required.
1.  **please back up your copy of .env**. Easiest way to simply rename your env . Alternatively if you have too much going on, you can start with a fresh repo clone.
1. Pull down this branch
1. Once VS Code detects `devcontainer.json` it will ask you to `rebuild this in devcontainers`. Alternatively, you can open the command line palette and choose `devcontainers: rebuild in container`. See this tutorial https://code.visualstudio.com/docs/devcontainers/tutorial
![image](https://github.com/forem/forem/assets/15793250/607cbbd4-2419-44d3-8472-c0825126fcc4)
1. It should take no more than 10 minutes to set it up.
![image](https://github.com/forem/forem/assets/15793250/157a58a4-41f4-4114-bfce-da1380151d8c).

## Added tests?

## Added to documentation?

## [optional] Are there any post deployment tasks we need to perform?
n/a
